Keep the ringrtc code out of the main WebRTC tree

This is an attempt to keep the ringrtc code out of the main WebRTC
tree, yet still leverage the WebRTC gn/ninja build system.

Step 1 -- Add a generic //ringrtc/BUILD.gn file that depends on an out
of tree BUILD.gn file using an absolute path.

This file includes a <never-checked-in> generated file,
ringrtc-root-dir.gni, that contains the absolute path of the actual
ringrtc source code.  The ringrtc-root-dir.gni file is generated at
the time the work space is created.

Step 2 -- Update the top level //BUILD.gn file to depend on the minimal
//ringrtc/BUILD.gn file.
---
 BUILD.gn         |    1 +
 ringrtc/BUILD.gn |    7 +++++++
 2 files changed, 8 insertions(+)
 create mode 100644 ringrtc/BUILD.gn

diff --git a/BUILD.gn b/BUILD.gn
index 31cf726ee4ae..9b02e1cafecd 100644
--- a/BUILD.gn
+++ b/BUILD.gn
@@ -420,6 +420,7 @@ if (!build_with_chromium) {
       "pc:peerconnection",
       "pc:rtc_pc",
       "pc:rtc_pc_base",
+      "ringrtc",
       "rtc_base",
       "sdk",
       "video",
diff --git a/ringrtc/BUILD.gn b/ringrtc/BUILD.gn
new file mode 100644
index 000000000000..46ab0fc153a7
--- /dev/null
+++ b/ringrtc/BUILD.gn
@@ -0,0 +1,7 @@
+import("ringrtc-src-dir.gni")
+
+group("ringrtc") {
+  public_deps = [
+    "${ringrtc_src}:ringrtc",
+  ]
+}
